rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -214,6 +213,17 @@ static sys_net_error get_last_error(bool is_blocking, int native_error = 0)
 		ERROR_CASE(ECONNRESET);
 		ERROR_CASE(EADDRINUSE);
 		ERROR_CASE(EISCONN);
+		ERROR_CASE(EINVAL);
+		ERROR_CASE(EINTR);
+		ERROR_CASE(EFAULT);
```

Comment:
EFAULT cannot be handled by operating systems in the way we want it. EINTR is just unimplemented on our side related to sys_net_abort being unimplemented, shouldnt be here in the switch.
I think the only error which actually makes sense here is EINVAL, only that.

Reply:
If we're converting these to SYS_NET_ errors (which appears to be the intent of this function), then we should be handling ALL of the errors that might reach this function.
If as you say the OS doesn't generate these in the way that we want, well then we should never be calling this function passing in such an error code from the OS, we should have captured it and reprocessed it earlier.

Reply:
really drop that EFAULT though. its just silencing possible bugs in our code, PS3 memory can be not allocated/memory protected on rpcs3 yet return code should be CELL_OK because its may need treatment in access violation handler.

Reply:
if we really wanted to implement EFAULT we had be doing stuff like vm::try_access which obeys our os3 page information.

Reply:
Maybe I'm not understanding this function.  For my addition, where is the silencing?
My understanding is that my change on this particular line would transform a Windows valued WSAEFAULT (10014) -> SYS_NET_EFAULT (14), and provide this back to the caller where it could be returned to the PS3 app.

Agree that there's possibly some vm memory fault issues that are potentially unresolved, but I'm not sure how this change would silence this situation.
It would just allow the PS3 application to actually understand better what is going on.. currently if an EFAULT does occur, then the PS3 application will get an error value of 1004... it won't have a clue what that means.. and it might even just have a switch case default which says 'unknown code'... try again (softlock ensues).. rather than saying 'ahh EFAULT, let's report network error and move on without network'

Reply:
atleast unknown errors are logged properly, maybe even throw an exception on unkown errors in the future but thats when the emulation code of sys_net is more matured. In which case EFAULT from the operating system should throw.

Reply:
I did notice that we had some throws in the sys_net code.  I thought we were against C++ exceptions (isn't that why we can't use mainline yaml?).

Reply:
I mean stuff like fmt::throw_exception which arent really throw.